File: src/mod11_gringotts_concurrency/lesson13_fork_join/java/Transfer.java
package mod11_gringotts_concurrency.lesson13_fork_join.java;

public class Transfer {

    private final BankAccount srcAccount;
    private final BankAccount targetAccount;
    private final int amount;

    public Transfer(BankAccount srcAccount, BankAccount targetAccount, int amount) {
        this.srcAccount = srcAccount;
        this.targetAccount = targetAccount;
        this.amount = amount;
    }
    public void execute(){
        srcAccount.transferFunds(targetAccount,amount);
    }
}

File: src/mod11_gringotts_concurrency/lesson13_fork_join/kotlin/Transfer.kt
package mod11_gringotts_concurrency.lesson13_fork_join.kotlin

class Transfer(
    private val srcAccount: BankAccount,
    private val targetAccount: BankAccount,
    private val amount: Int
) {
    fun execute() {
        srcAccount.transferFunds(targetAccount, amount)
    }

}